executor/metric_reader.go

```diff
@@ -243,3 +243,72 @@ func (e *MetricSummaryRetriever) genMetricQuerySQLS(name, startTime, endTime str
 	}
 	return sqls
 }
+
+// MetricSummaryByLabelRetriever uses to read metric detail data.
+type MetricSummaryByLabelRetriever struct {
+	dummyCloser
+	table     *model.TableInfo
+	extractor *plannercore.MetricTableExtractor
+	retrieved bool
+}
+
+func (e *MetricSummaryByLabelRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	if e.retrieved || e.extractor.SkipRequest {
+		return nil, nil
+	}
+	e.retrieved = true
+	totalRows := make([][]types.Datum, 0, len(infoschema.MetricTableMap))
+	quantiles := []float64{1, 0.999, 0.99, 0.90, 0.80}
+	tps := make([]*types.FieldType, 0, len(e.table.Columns))
+	for _, col := range e.table.Columns {
+		tps = append(tps, &col.FieldType)
+	}
+	startTime := e.extractor.StartTime.Format(plannercore.MetricTableTimeFormat)
+	endTime := e.extractor.EndTime.Format(plannercore.MetricTableTimeFormat)
+	for name, def := range infoschema.MetricTableMap {
+		sqls := e.genMetricQuerySQLS(name, startTime, endTime, def.Quantile, quantiles, def)
```

Comment:
The `e.genMetricQuerySQLS` method accepts the `def`, so I think the `def.Quantile` is unnecessary.

Reply:
great, done.